persistence/persistence-mybatis/src/main/java/org/cbioportal/persistence/mybatis/AlterationMyBatisRepository.java

```diff
@@ -63,10 +61,8 @@
             return Collections.emptyList();
         }
 
-        List<Integer> internalPatientIds = alterationCountsMapper.getPatientInternalIds(molecularProfileCaseIdentifiers);
-
```

Comment:
The fetching of patient ids was extracted from the mapper to the repository in order to reuse the sample ids between the   mutation and cna count routines. Now, you reintroduced the double and therefore redundant evaluation of sample/case internal ids. I personally think that the new mechanism does not have to be removed. All that is needed is to update the mapper.xml for _getSampleInternalIds_ and _getPatientInternalIds_. I am wondering why you removed the new mechanism is favor over the old situation?

Reply:
Updating _getSampleInternalIds_ and _getPatientInternalIds_ doesn't resolve the issue. We need to include molecular profile ids while querying for mutation and cna counts.